provider/zone_type_filter_test.go

```diff
@@ -69,3 +69,39 @@ func TestZoneTypeFilterMatch(t *testing.T) {
 		assert.Equal(t, tc.matches, zoneTypeFilter.Match(tc.zone))
 	}
 }
+
+func TestZoneTypeFilterMatch(t *testing.T) {
+	publicZone := "public"
+	privateZone := "private"
+
+	for _, tc := range []struct {
```

Comment:
It seems the code is more or less the same as the one above (in `TestZoneTypeFilterMatchAWS`). What if we would just wrap it into one function? So in function `TestZoneTypeFilterMatch` we test everything, meaning

`publicZone := "public"`, 
`privateZone := "private"`,  
`publicZone := &route53.HostedZone{Config: &route53.HostedZoneConfig{PrivateZone: aws.Bool(false)}}` and
`privateZone := &route53.HostedZone{Config: &route53.HostedZoneConfig{PrivateZone: aws.Bool(true)}}`

so we can avoid the duplicate code and remove the extra `TestZoneTypeFilterMatchAWS` 🙂

Reply:
Good point. I refactored these tests

Reply:
@njuettner Any chance of a rereview?

Reply:
Sorry a bit late 😞  mostly due to other priorities. I'll approve it 👍🏻 , thank you for fixing it 🚀